sky/data/storage.py

```diff
@@ -1253,9 +1253,11 @@ def _delete_gcs_bucket(self, bucket_name: str) -> None:
             with backend_utils.safe_console_status(
                     f'[bold cyan]Deleting [green]bucket {bucket_name}'):
                 if num_files >= _GCS_RM_MAX_OBJS:
-                    remove_obj_command = f'gsutil -m rm -a gs://{bucket_name}/*'
+                    remove_obj_command = ('gsutil -m rm -r'
+                                          f' gs://{bucket_name}')
                     subprocess.check_output(remove_obj_command.split(' '))
-                bucket.delete(force=True)
+                else:
+                    bucket.delete(force=True)
```

Comment:
Out of curiosity, why do we need to separate the two cases?

Reply:
We can have both cases together, but the 2nd one is slightly faster imo.

Reply:
Does the speed-up of the 2nd one amortize the number of files check? I am guessing the `gsutil du` will take a long time as well, but if we only use the 1st command, this step can be skipped.
```
        num_files = subprocess.check_output(
            f'gsutil du gs://{bucket_name} | wc -l', shell=True)
        num_files = int(num_files)
```

Should we get some profiling numbers for the two methods if `bucket.delete` is faster, and leave a comment in the code to avoid future confusion?

Reply:
You are right `du` is slow for large buckets. It is better to straight up delete. Fixed code. No need to `bucket.delete()` anymore.